google-cloud-firestore/lib/google/cloud/firestore.rb

```diff
@@ -21,6 +21,381 @@ module Cloud
     ##
     # # Cloud Firestore
     #
+    # Cloud Firestore is a NoSQL document database built for automatic scaling,
+    # high performance, and ease of application development. While the Cloud
+    # Firestore interface has many of the same features as traditional
+    # databases, as a NoSQL database it differs from them in the way it
+    # describes relationships between data objects.
+    #
+    # For more information about Cloud Firestore, read the [Cloud
+    # Firestore Documentation](https://cloud.google.com/firestore/docs/).
+    #
+    # The goal of google-cloud is to provide an API that is comfortable to
+    # Rubyists. Authentication is handled by {Google::Cloud#spanner}. You can
+    # provide the project and credential information to connect to the Cloud
+    # Spanner service, or if you are running on Google Compute Engine this
+    # configuration is taken care of for you. You can read more about the
+    # options for connecting in the [Authentication
+    # Guide](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/guides/authentication).
+    #
+    # ## Adding data
+    #
+    # Cloud Firestore stores data in Documents, which are stored in Collections.
+    # Cloud Firestore creates collections and documents implicitly the first
+    # time you add data to the document. You do not need to explicitly create
+    # collections or documents. (For more information, see [Adding Data to Cloud
+    # Firestore](https://cloud.google.com/firestore/docs/manage-data/add-data).
+    #
+    # To create or overwrite a single document, use {Firestore::Client#doc} to
+    # obtain a document reference. (This does not create a document in Cloud
+    # Firestore.) Then, call {Firestore::DocumentReference#set} to create the
+    # document or overwrite an existing document:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a document reference
+    # nyc_ref = firestore.doc "cities/NYC"
+    #
+    # nyc_ref.set({ name: "New York City" }) # Document created
+    # ```
+    #
+    # When you use this combination of `doc` and set` to create a new document,
+    # you must specify an ID for the document. Sometimes there isn't a
```

Comment:
Calling `CollectionReference#doc` without an argument will also generate a random ID.

Reply:
Hmmmmm. Which should we show?

Reply:
How about something like this?

> When you use this combination of `doc` and set` to create a new document, you can specify an ID for the document. If you do not have a meaningful ID for the document, you can omit the ID and Firestore auto-generate an ID for you. You can also perform both of these operations by calling {Firestore::CollectionReference#add}, which will fail if the document already exists.